packages/dataviews/src/components/dataviews-pagination/index.tsx

```diff
@@ -40,38 +40,35 @@ function DataViewsPagination() {
 					spacing={ 2 }
 					className="dataviews-pagination__page-selection"
 				>
-					{ createInterpolateElement(
-						sprintf(
-							// translators: %s: Total number of pages.
-							_x( 'Page <CurrentPageControl /> of %s', 'paging' ),
-							totalPages
-						),
-						{
-							CurrentPageControl: (
-								<SelectControl
-									aria-label={ __( 'Current page' ) }
-									value={ view.page?.toString() }
-									options={ Array.from(
-										Array( totalPages )
-									).map( ( _, i ) => {
-										const page = i + 1;
-										return {
-											value: page.toString(),
-											label: page.toString(),
-										};
-									} ) }
-									onChange={ ( newValue ) => {
-										onChangeView( {
-											...view,
-											page: +newValue,
-										} );
-									} }
-									size="compact"
-									__nextHasNoMarginBottom
-								/>
-							),
-						}
-					) }
+					<SelectControl
+						aria-label={ __( 'Current page' ) }
+						value={ view.page?.toString() }
+						options={ Array.from( Array( totalPages ) ).map(
+							( _, i ) => {
+								const page = i + 1;
+								return {
+									value: page.toString(),
+									label:
+										__( 'Page ' ) +
+										page.toString() +
+										sprintf(
+											// translators: %s: Total number of pages.
+											_x( ' of %s', 'paging' ),
+											totalPages
+										),
```

Comment:
This configuration may cause problems with localization. Perhaps we could configure it like this:

```javascript
sprintf(
	// translators: 1: Current page number, 2: Total number of pages
	__( 'Page %1$s of %2$s' ),
	page.toString(),
	totalPages
)
```